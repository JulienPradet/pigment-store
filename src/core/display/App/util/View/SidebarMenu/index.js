import React from 'react'
import classnames from 'classnames'
import styles from './sidebar.css'

export const SidebarMenu = ({children}) => <div>{children}</div>
SidebarMenu.displayName = 'SidebarMenu.SidebarMenu'

export const MenuTitle = ({children}) => <h2 className={styles.title}>{children}</h2>
MenuTitle.displayName = 'SidebarMenu.MenuTitle'

export const Container = ({children}) => <ul className={styles.list}>
  {children}
</ul>
Container.displayName = 'SidebarMenu.Container'

export const Item = ({isActive, fade, children}) => <li className={classnames(styles.item, {[styles.activeItem]: isActive, [styles.fade]: fade})}>
  {children}
</li>
Item.displayName = 'SidebarMenu.Item'

export const Search = ({search, onChange}) => {
  return <div className={styles.search}>
    <input type='text' value={search} onChange={(e) => onChange(e.target.value)} placeholder='Search...' />
    <button onClick={() => onChange('')}>
      <i className='material-icons'>
        {search ? 'close' : 'search'}
      </i>
    </button>
  </div>
}
Search.displayName = 'SidebarMenu.Search'
