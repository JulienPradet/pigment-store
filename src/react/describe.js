import describe from './definition/Component'

export default describe
