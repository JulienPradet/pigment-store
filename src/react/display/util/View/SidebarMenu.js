import React from 'react'

export const Container = ({children}) => <ul>
  {children}
</ul>

export const Item = ({children}) => <li>
  {children}
</li>
