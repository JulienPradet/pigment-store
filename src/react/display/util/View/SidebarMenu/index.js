import React from 'react'
import classnames from 'classnames'
import styles from './sidebar.css'

export const MenuTitle = ({children}) => <h2 className={styles.title}>{children}</h2>

export const Container = ({children}) => <ul className={styles.list}>
  {children}
</ul>

export const Item = ({isActive, fade, children}) => <li className={classnames(styles.item, {[styles.activeItem]: isActive, [styles.fade]: fade})}>
  {children}
</li>

export const Search = ({search, onChange}) => {
  return <div className={styles.search}>
    <input type='text' value={search} onChange={(e) => onChange(e.target.value)} placeholder='Search...' />
    <button onClick={() => onChange('')}>
      <i className='material-icons'>
        {search ? 'close' : 'search'}
      </i>
    </button>
  </div>
}
