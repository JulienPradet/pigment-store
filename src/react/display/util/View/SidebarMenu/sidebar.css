.title {
  display: block;
  font-size: 2em;
  margin: 0;
  padding: 0.5em;
}

.title, .title a {
  background: rgb(85, 119, 255);
  color: white;
  text-decoration: none;
}

.list {
  margin: 0;
  padding: 0;
}

.list:first-of-type {
  border-top: 1px solid rgb(200, 200, 200);
}

.list .list:first-of-type {
  border-top: 0;
}

.item {
  display: block;
  list-style: none;
}

.item a {
  display: block;
  padding: 1em;
  border-bottom: 1px solid rgb(200, 200, 200);
  color: #444;
  text-decoration: none;
}

.item mark {
  background: #444;
  color: #fff;
}

.item input:only-child {
  position: relative;
  padding: 1em;
  box-sizing: border-box;
  width: 100%;
  border-width: 0;
  border-bottom: 1px solid rgb(200, 200, 200);
}

.activeItem {
  background: rgba(255, 255, 255, 0.5);
}

.activeItem > a {
  font-weight: bold;
}

.list .list .item a {
  padding-left: 2em;
}

.list .list .list .item a {
  padding-left: 3em;
}
