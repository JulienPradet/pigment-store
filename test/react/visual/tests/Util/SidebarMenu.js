import React from 'react'
import describe from 'pigment-store/react/describe'
import {SidebarMenu, MenuTitle, Item, Search, Container} from '../../../../../src/core/display/App/util/View/SidebarMenu/index'

const onSearchChange = (newSearch) => {
  console.log(newSearch)
}

export default describe('SidebarMenu', SidebarMenu)
  .featureJsx(
    'Default',
    () => (
      <SidebarMenu>
        <MenuTitle><a href='/'>Pigment Store</a></MenuTitle>
        <Item>
          <Search search='Current search' onChange={onSearchChange} />
        </Item>
        <Item>
          <Container>
            <Item>
              <a href='/menu'>About</a>
            </Item>
          </Container>
        </Item>
      </SidebarMenu>
    )
  )
  .featureJsx(
    'With sub categories',
    () => (
      <SidebarMenu>
        <MenuTitle><a href='/'>Pigment Store</a></MenuTitle>
        <Item>
          <Search search='Current search' onChange={onSearchChange} />
        </Item>
        <Item>
          <Container>
            <Item isActive>
              <a href='/category'>Category</a>
              <Container>
                <Item>
                  <a href='/category/section1'>Section 1</a>
                </Item>
                <Item>
                  <a href='/category/section2'>Section 2</a>
                </Item>
                <Item>
                  <a href='/category/section3'>Section 3</a>
                </Item>
              </Container>
            </Item>
          </Container>
        </Item>
      </SidebarMenu>
    )
  )
