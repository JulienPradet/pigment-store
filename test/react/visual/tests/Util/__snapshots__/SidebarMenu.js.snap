exports[`SidebarMenu Default 1`] = `
<div>
  <h2
    className="title">
    <a
      href="/">
      Pigment Store
    </a>
  </h2>
  <li
    className="item">
    <div
      className="search">
      <input
        onChange={[Function]}
        placeholder="Search..."
        type="text"
        value="Current search" />
      <button
        onClick={[Function]}>
        <i
          className="material-icons">
          close
        </i>
      </button>
    </div>
  </li>
  <li
    className="item">
    <ul
      className="list">
      <li
        className="item">
        <a
          href="/menu">
          About
        </a>
      </li>
    </ul>
  </li>
</div>
`;

exports[`SidebarMenu With sub categories 1`] = `
<div>
  <h2
    className="title">
    <a
      href="/">
      Pigment Store
    </a>
  </h2>
  <li
    className="item">
    <div
      className="search">
      <input
        onChange={[Function]}
        placeholder="Search..."
        type="text"
        value="Current search" />
      <button
        onClick={[Function]}>
        <i
          className="material-icons">
          close
        </i>
      </button>
    </div>
  </li>
  <li
    className="item">
    <ul
      className="list">
      <li
        className="item activeItem">
        <a
          href="/category">
          Category
        </a>
        <ul
          className="list">
          <li
            className="item">
            <a
              href="/category/section1">
              Section 1
            </a>
          </li>
          <li
            className="item">
            <a
              href="/category/section2">
              Section 2
            </a>
          </li>
          <li
            className="item">
            <a
              href="/category/section3">
              Section 3
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </li>
</div>
`;
